alarm/models.py

```diff
@@ -13,9 +16,57 @@ def __str__(self):
 
 
 class Coin(models.Model):
-    phone = models.ForeignKey(Phone, on_delete=models.CASCADE)
-    coin_abbreviation = models.CharField(max_length=255)
+    phone = models.ForeignKey(Phone, on_delete=models.CASCADE, default=None)
+    abbreviation = models.CharField(max_length=255, default=None)
     threshold = models.DecimalField(max_digits=10, decimal_places=2)
 
     def __str__(self):
-        return str(self.coin_abbreviation)
+        return str(self.abbreviation)
+
+    @property
+    def get_last_candle(self):
+        """Returns the last candle object associated with this coin."""
+        return self.candle.filter(coin=self).last()
+
+    @property
+    def last_high_price(self):
+        """Returns the high price of the last candle object associated with this coin."""
+        last_candle = self.get_last_candle
+        if last_candle:
+            return last_candle.high_price
+        else:
+            return None
+
+    @property
+    def last_low_price(self):
+        """Returns the low price of the last candle object associated with this coin."""
+        last_candle = self.get_last_candle
+        if last_candle:
+            return last_candle.low_price
+        else:
+            return None
+
+    def update_or_create_candles(self, current_candle_high_price, current_candle_low_price):
+        Candle.objects.update_or_create(
+            coin=self,
+            defaults={
+                'high_price': current_candle_high_price,
+                'low_price': current_candle_low_price,
+            },
+        )
+
+    def clean(self):
+        super().clean()
+        valid_list_of_symbols = get_binance_list_of_coin_names()
```

Comment:
same here, function return list_of_coin_names but the variable is list_of_symbols

Reply:
I'd call it list_of_binance_coins_abbreviations

Reply:
++